### ISSUE TITLE
EPO.MINIMIZE and EPO.MAXIMIZE

### ISSUE THREAD

**Issue by IsmailBData** [khalil-research/PyEPO#60]
Hello,

I would like to understand this better. Does this mean that if I have a minimization problem, I should use the code inside the if block, and if I have a maximization problem, I should use the code inside the else block?
I’m asking because you mentioned in this issue that they are the same: [[https://github.com/khalil-research/PyEPO/issues/56#issuecomment-3483492270](url)](https://github.com/khalil-research/PyEPO/issues/56#issuecomment-3483492270)

`        
if module.optmodel.modelSense == EPO.MINIMIZE:
            loss = - obj + 2 * torch.einsum("bi,bi->b", cp, w) - z.squeeze(dim=-1)
elif module.optmodel.modelSense == EPO.MAXIMIZE:
            loss = obj - 2 * torch.einsum("bi,bi->b", cp, w) + z.squeeze(dim=-1)`

**Comment by LucasBoTang** [comment]
The difference is based on the sign.

**Comment by LucasBoTang** [comment]
Just to clarify: the two branches are not literally identical expressions. They are mathematically equivalent up to the sign convention needed to handle minimization vs. maximization, but writing them as two branches in code does change the sign of terms. 

Also, Issue #56 does not state that the two branches are identical. You can clearly see that the signs differ in the expressions.
